app/models/publications/monthly_statistics/by_sex.rb

```diff
@@ -46,40 +46,89 @@ def formatted_group_query
         }
 
         group_query_excluding_deferred_offers.map do |item|
-          sex, status = item[0]
-          count = item[1]
-          counts[sex]&.merge!({ status => count })
+          counts[item['sex']]&.merge!({ item['status'] => item['count'] })
         end
 
         group_query_for_deferred_offers.map do |item|
-          sex, status = item[0]
-          count = item[1]
-          counts[sex]&.merge!({ status => count })
+          counts[item['sex']]&.merge!({ item['status_before_deferral'] => item['count'] })
         end
 
         counts
       end
 
       def group_query_for_deferred_offers
-        group_query(recruitment_cycle_year: RecruitmentCycle.previous_year)
-          .where(status: :offer_deferred)
-          .count
+        group_query(
+          cycle: RecruitmentCycle.previous_year,
+          status_attribute: :status_before_deferral,
+        )
       end
 
       def group_query_excluding_deferred_offers
-        group_query(recruitment_cycle_year: RecruitmentCycle.current_year)
-          .where.not(status: :offer_deferred)
-          .count
+        group_query
       end
 
-      def group_query(recruitment_cycle_year: RecruitmentCycle.current_year)
-        ApplicationChoice
-          .joins(:application_form)
-          .where(application_forms: { recruitment_cycle_year: recruitment_cycle_year })
-          .where.not(
-            application_forms: ApplicationForm.select(:previous_application_form_id).where.not(previous_application_form_id: nil),
-          )
-          .group("application_forms.equality_and_diversity->'sex'", 'status')
+      def group_query(
+        cycle: RecruitmentCycle.current_year,
+        status_attribute: :status
+      )
+        without_subsequent_applications_query =
+          "AND (
+            NOT EXISTS (
+              SELECT 1
+              FROM application_forms
+              AS subsequent_application_forms
+              WHERE application_forms.id = subsequent_application_forms.previous_application_form_id
+            )
+          )"
+        with_statuses =
+          if status_attribute.to_s == 'status_before_deferral'
```

Comment:
Given that we only call this from inside this class, perhaps we could leave as symbol?